scripts.js

```diff
@@ -0,0 +1,81 @@
+// Load countries from JSON file, once completed use displayCountries to render them
+const loadCountries = async () => {
+  const grid = document.getElementById("countries-grid");
+
+  try {
+    const response = await fetch("./CountriesData.json");
+    const countries = await response.json();
+    displayCountries(countries);
+  } catch (error) {
+    console.error("Error loading countries:", error);
+  }
+};
+
+const displayCountries = (countries) => {
+  const grid = document.getElementById("countries-grid");
+  grid.innerHTML = ""; // Reset the grid
+
+  countries.forEach((country) => {
+    const countryElement = document.createElement("a");
+    countryElement.href = "#";
+    countryElement.className = "country scale-effect";
+    countryElement.setAttribute("data-country-name", country.name);
+
+    countryElement.innerHTML = `
+        <div class="country-flag">
+          <img
+            src="${country.flag}"
+            alt="${country.name} Flag"
+          />
+        </div>
+        <div class="country-info">
+          <h2 class="country-title">${country.name}</h2>
+          <ul class="country-brief">
+            <li><strong>Population: </strong>${country.population}</li>
+            <li><strong>Region: </strong>${country.region}</li>
+            <li><strong>Capital: </strong>${country.capital}</li>
+          </ul>
+        </div>
+      `;
+
+    // Append the country element to the grid
+    grid.appendChild(countryElement);
+  });
+};
+
+const filterCountries = async (region) => {
+  try {
+    const response = await fetch("./CountriesData.json");
+    const countries = await response.json();
```

Comment:
This function should only do 1 thing and this is to filter the countries.
Extract the get data logic into separate function